pennylane/devices/default_clifford.py

```diff
@@ -519,58 +692,322 @@ def _measure_state(self, tableau_simulator, circuit, global_phase):
                     np.hstack((z2x, z2z, z_signs.reshape(-1, 1))),
                 )
             ).astype(int)
-            if pl_tableau.shape == (0, 1) and circuit.num_wires:
+            if pl_tableau.shape == (0, 1) and len(wires):
                 return np.array([[1.0, 0.0, 0.0], [0.0, 1.0, 0.0]])
             return pl_tableau
 
         state = qml.math.array(tableau_simulator.state_vector(endian="big"))
-        if state.shape == (1,) and circuit.num_wires:
+        if state.shape == (1,) and len(wires):
             # following is faster than using np.eye(length=1, size, index)
-            state = qml.math.zeros(2**circuit.num_wires, dtype=complex)
+            state = qml.math.zeros(2 ** len(wires), dtype=complex)
             state[0] = 1.0 + 0.0j
-        return state * qml.matrix(global_phase)
+        return state * qml.matrix(global_phase)[0][0]
 
     @staticmethod
-    def _measure_density_matrix(tableau_simulator):
-        """Measure the density matrix from the state of simulator device."""
-        state_vector = qml.math.array(tableau_simulator.state_vector(endian="big"))
-        return qml.math.einsum("i, j->ij", state_vector, state_vector)
+    def _measure_expectation(meas, tableau_simulator, **kwargs):
+        """Measure the expectation value with respect to the state of simulator device."""
+        # Get the observable for the expectation value measurement
+        (coeffs, paulis), _ = _convert_op_to_linear_comb(meas.obs), kwargs
+
+        expecs = qml.math.zeros_like(coeffs)
+        for idx, (pauli, wire) in enumerate(paulis):
+            pauli_term = ["I"] * max(np.max(list(wire)) + 1, tableau_simulator.num_qubits)
+            for op, wr in zip(pauli, wire):
+                pauli_term[wr] = op
+            stim_pauli = stim.PauliString("".join(pauli_term))
+            expecs[idx] = tableau_simulator.peek_observable_expectation(stim_pauli)
+
+        return qml.math.dot(coeffs, expecs)
+
+    def _measure_variance(self, meas, tableau_simulator, **kwargs):
+        """Measure the variance with respect to the state of simulator device."""
+        meas_obs, _ = qml.operation.convert_to_opmath(meas.obs), kwargs
```

Comment:
your pylint solution made me smile but this would do just fine and probably confuse people less
```suggestion
    def _measure_variance(self, meas, tableau_simulator, **_):
        """Measure the variance with respect to the state of simulator device."""
        meas_obs = qml.operation.convert_to_opmath(meas.obs)
```